File: samplestore/src/main/java/com/stripe/samplestore/ItemDivider.java
package com.stripe.samplestore;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.drawable.Drawable;
import android.view.View;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

class ItemDivider extends RecyclerView.ItemDecoration {

    @NonNull private final Drawable divider;

    /**
     * Custom divider will be used in the list.
     */
    ItemDivider(@NonNull Context context, @DrawableRes int resId) {
        divider = ContextCompat.getDrawable(context, resId);
    }

    @Override
    public void onDraw(@NonNull Canvas c, @NonNull RecyclerView parent,
                       @NonNull RecyclerView.State state) {
        final int start = parent.getPaddingStart();
        final int end = parent.getWidth() - parent.getPaddingEnd();

        int childCount = parent.getChildCount();
        for (int i = 0; i < childCount; i++) {
            View child = parent.getChildAt(i);

            RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child.getLayoutParams();

            int top = child.getBottom() + params.bottomMargin;
            int bottom = top + divider.getIntrinsicHeight();

            divider.setBounds(start, top, end, bottom);
            divider.draw(c);
        }
    }
}

File: samplestore/src/main/java/com/stripe/samplestore/ItemDivider.kt
package com.stripe.samplestore

import android.content.Context
import android.graphics.Canvas
import android.graphics.drawable.Drawable
import androidx.annotation.DrawableRes
import androidx.core.content.ContextCompat

/**
 * Custom divider will be used in the list.
 */
internal class ItemDivider(
    context: Context,
    @DrawableRes resId: Int
) : androidx.recyclerview.widget.RecyclerView.ItemDecoration() {

    private val divider: Drawable = ContextCompat.getDrawable(context, resId)!!

    override fun onDraw(c: Canvas, parent: androidx.recyclerview.widget.RecyclerView, state: androidx.recyclerview.widget.RecyclerView.State) {
        val start = parent.paddingStart
        val end = parent.width - parent.paddingEnd

        val childCount = parent.childCount
        for (i in 0 until childCount) {
            val child = parent.getChildAt(i)

            val params = child.layoutParams as androidx.recyclerview.widget.RecyclerView.LayoutParams

            val top = child.bottom + params.bottomMargin
            val bottom = top + divider.intrinsicHeight

            divider.setBounds(start, top, end, bottom)
            divider.draw(c)
        }
    }
}
